pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -103,3 +103,33 @@ func (r *ReconcileCluster) updateStatus(cluster *synv1alpha1.Cluster) error {
 	return r.client.Status().Update(context.TODO(), cluster)
 
 }
+
+func (r *ReconcileCluster) addLabels(cluster *synv1alpha1.Cluster) error {
+
+	_, err := r.fetchInstance(cluster, types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace})
+	if err != nil {
+		return err
+	}
+
+	if cluster.ObjectMeta.Labels == nil {
+		cluster.ObjectMeta.Labels = make(map[string]string)
+	}
+	cluster.ObjectMeta.Labels["tenant"] = cluster.Spec.TenantRef.Name
+	return r.client.Update(context.TODO(), cluster)
```

Comment:
I'd recommend to only update the object once at the end of the reconcile run to avoid multiple calls to the API and often re-queueing of the object.

Reply:
See todos in code.

Reply:
I can't currently set the url in gitRepoURL, as the logic nessary for that will be implemented in SYn-25. So currently it will set it to "dummy".